calculator.rb

```diff
@@ -0,0 +1,149 @@
+  operation = gets.chomp.strip
+end
+
+# Gets first number from user (string)
+print "What is your first number: "
+first_num = gets.chomp
+
+# Prints error message if not a number
+while first_num != "0" && first_num.to_i == 0
+  print "Whoops! That is not a number. Please enter a number: "
+  first_num = gets.chomp
+end
+
+# Gets second number from user (string)
+print "What is your second number: "
+second_num = gets.chomp
+
+# Prints error message if not a number
+while second_num != "0" && second_num.to_i == 0
+  print "Whoops! That is not a number. Please enter a number: "
+  second_num = gets.chomp
+end
+
+# Asks if user wants answer as an int or float
+print "Would you like your answer as an integer or float? (type \"i\" for integer and \"f\" for float): "
+num_type = gets.chomp
+
+# converts the input numbers to either an integer or float or error message
+if num_type == "i"
+  first_num = first_num.to_i
+  second_num = second_num.to_i
+elsif num_type == "f"
+  first_num = first_num.to_f
+  second_num = second_num.to_f
+else
+  print "That was not a valid command. Please try again: "
+  num_type = gets.chomp
+end
+
+
+# CALCULATIONS------------------------------------------------------------------
+
+# ADDITION
+if (operation == "add") || (operation == "+")
+  puts "#{ first_num } + #{ second_num } = #{ first_num + second_num }"
+
+# SUBTRATION
+elsif (operation == "sub") || (operation == "-")
+
+  # Asks to confirm order
+  print "Would you like to do: #{ first_num } - #{ second_num } (type \"1\") or #{ second_num } - #{ first_num } (type\"2\")? "
+  order_of_oper = gets.chomp.to_i
+
+  # User input error check
+  while (order_of_oper != 1) && (order_of_oper != 2)
+    print "That's not a valid command. Please enter a \"1\" or a \"2\": "
+    order_of_oper = gets.chomp.to_i
+  end
+
+  # Does calculation in correct order
+  if (order_of_oper == 1)
+    puts "#{ first_num } - #{ second_num } = #{ first_num - second_num }"
+  elsif (order_of_oper == 2)
+    puts "#{ second_num } - #{ first_num } = #{ second_num - first_num }"
+  end
+
+# MULTIPLICATION
+elsif (operation == "mult") || (operation == "*")
+  puts "#{ first_num } * #{ second_num } = #{ first_num * second_num }"
+
+# DIVISION
+elsif (operation == "div") || (operation == "/")
+
+  # Asks to confirm order
+  print "Would you like to do: #{ first_num } / #{ second_num } (type \"1\") or #{ second_num } / #{ first_num } (type\"2\")? "
+  order_of_oper = gets.chomp.to_i
+
+  # User input error check
+  while (order_of_oper != 1) && (order_of_oper != 2)
+    print "That's not a valid command. Please enter a \"1\" or a \"2\": "
+    order_of_oper = gets.chomp.to_i
+  end
+
+  # Does calculation in correct order
+  # Also checks to make sure user isn't trying to divide by zero
+  # Also gives remainder
+  if (order_of_oper == 1)
+    if (second_num == 0)
+      puts "You cannot divide by zero. Sorry!"
+    else
+      if (first_num.class == Fixnum)
```

Comment:
I love the check on this line; Ruby's math varies by object class, and controlling for that is fantastic.

In the current setup, if `second_num` is a Float (2.5), but `first_num` is a Fixnum (2), you'll end up with something like `2 / 2.5 = 0.8 with a remainder of 2.0`.  My advice would be to try checking for the reverse state and checking both variables, like `if (first_num.class == Float || second_num.class == Float)`.